src/nupic/algorithms/Connections.cpp

```diff
@@ -628,20 +618,20 @@ void Connections::load(std::istream &inStream) {
 
 CellIdx Connections::numCells() const { return (CellIdx)cells_.size(); }
 
-UInt Connections::numSegments() const {
-  return (UInt)(segments_.size() - destroyedSegments_.size());
+Segment Connections::numSegments() const {
+  return (Segment)(segments_.size() - destroyedSegments_.size());
```

Comment:
Because of this typecase: `segments_.size()` must be `<= numeric_limits<Segment>::max()`

Reply:
here, using Segment is a bit confusing, should we rather use size_t for anything that returns a size of something? (and check it actually fits where used?)

Reply:
I'll improve this in #357